<?xml version="1.0"?>
<robot name="hexapod">

  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.2"/>
      </geometry>
    </visual>
  </link>

  <link name="base_footprint">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
    </visual>
  </link>

  <joint name="base_to_footprint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>

  <link name="base_dms">
    <visual>
      <geometry>
        <box size="0.04 0.02 0.02"/>
      </geometry>
      <material name="blue">
        <color rgba="0 0 0.8 1.0"/>
      </material>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    </visual>
  </link>

  <joint name="base_to_dms" type="fixed">
    <parent link="base_link"/>
    <child link="base_dms"/>
    <origin rpy="0.0 0.0 1.571" xyz="0.04 0.0 0.05"/>
  </joint>

  <link name="base_laser">
    <visual>
      <geometry>
        <box size="0.1 0.05 0.2"/>
      </geometry>
      <material name="green">
        <color rgba="0 0.8 0 1.0"/>
      </material>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
    </visual>
  </link>

  <joint name="base_to_laser" type="fixed">
    <parent link="base_link"/>
    <child link="base_laser"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.1 0.0 0.1"/>
  </joint>

<!--
  <link name="imu">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <material name="green">
        <color rgba="0 0.8 0 1.0"/>
      </material>
      <origin rpy="0.0 0.0 1.57075" xyz="0.0 0.0 0.0"/>
    </visual>
  </link>

  <joint name="base_to_imu" type="fixed">
    <parent link="base_link"/>
    <child link="imu"/>
    <origin rpy="0.0 0.0 1.57075" xyz="0.0 0.05 0.25"/>
  </joint>
-->

</robot>
